### PR TITLE
Desktop, Mobile: Link resource Icon

### PR DESCRIPTION
**Pull request by CalebJohn**
This pull customizes the resource link icon based on the mimeType of the attached item. A screenshot will demonstrate what it does best
image

The icon will follow the current set theme, additionally the icons can be set using css by overwriting the fork-awesome css.
As you can see this only applies to resources so regular urls won't have an icon.

unfortunately I had a lot of trouble trying to get this working on mobile, which is why for now I have it disabled on mobile. @laurent22 if that doesn't work, could you help me out in getting mobile working? I tried copying the way that katex loads fonts at runtime, but I could never get the icons to render properly, even with that.

Based on [the forum post about this](https://discourse.joplinapp.org/t/set-of-icons-for-attachments/2813) maybe we should discuss exactly how we want resource/link icons to look before merging this.

### CODE REVIEW

**Comment by laurent22** [comment]
Thanks @CalebJohn, and the code looks good. I think we can indeed get it working on mobile so I'll take a look.

**Comment by laurent22** [comment]
Ok it's a bit complicated to get this working on mobile due to the font files that need to be embedded, so let's focus on desktop only for now.

I think the icons look a bit big, could you make them a bit smaller? In my tests 1.4em looked good. Also if you could add a bit more margin to the right of the icon (0.4em looked fine for me).

Finally, if I attach a .txt file, there's no icon for it, while I expect it's a commonly used file. Since we use Font Awesome lib to get the icon name, maybe it's a different name in Fork Awesome?

**Comment by CalebJohn** [comment]
I've changed them as you suggested, it's funny, I originally had 1.4em and 0.4em, but as I tweaked things around I settled on slightly different values. 

You were absolutely right about the txt issue, turns out that font awesome doesn't have a txt format, I added a special case for it and checked the rest of the icons to make ensure they're present in both.

In regards to the mobile issue, we can actually change the way we render icons back to the old method (inline svg) that tessus had added (which is still the fallback for mobile). I would only have to add ~11 inline svg icons. And we could use the font awesome names to clean up the code a bit!
What do you think? If it's alright with you I might add a commit here that would replace the font stuff.

**Comment by laurent22** [comment]
> In regards to the mobile issue, we can actually change the way we render icons back to the old method (inline svg) that tessus had added (which is still the fallback for mobile). I would only have to add ~11 inline svg icons. And we could use the font awesome names to clean up the code a bit!
> What do you think? If it's alright with you I might add a commit here that would replace the font stuff.

That sounds like a good solution and if you could figure out how to implement it that would be great. I guess we would only support the 11 most common file formats then, but that's probably more than enough.

**Comment by CalebJohn** [comment]
Fork awesome only really has that many so it shouldn't be an issue, additionally the library that we use (font-awesome-filetypes) only supports 12 (one of which is csv which isn't in fork awesome).

**Comment by CalebJohn** [comment]
I finished the updates we talked about. Also tested on mobile!

Here is all supported icons together
image

File: ReactNativeClient/lib/renderers/noteStyle.js [hunk 1]
```diff
@@ -105,10 +105,45 @@ module.exports = function(style, options) {
...
+		}
+		.fa-file-alt, .fa-file-csv {
+			-webkit-mask: url("data:image/svg+xml;utf8,<svg viewBox='0 0 1536 1992' xmlns='http://www.w3.org/2000/svg'><path d='M1468 380c37 37 68 111 68 164v1152c0 53-43 96-96 96H96c-53 0-96-43-96-96V96C0 43 43 0 96 0h896c53 0 127 31 164 68zm-444-244v376h376c-6-17-15-34-22-41l-313-313c-7-7-24-16-41-22zm384 1528V640H992c-53 0-96-43-96-96V128H128v1536h1280zM384 800c0-18 14-32 32-32h704c18 0 32 14 32 32v64c0 18-14 32-32 32H416c-18 0-32-14-32-32v-64zm736 224c18 0 32 14 32 32v64c0 18-14 32-32 32H416c-18 0-32-14-32-32v-64c0-18 14-32 32-32h704zm0 256c18 0 32 14 32 32v64c0 18-14 32-32 32H416c-18 0-32-14-32-32v-64c0-18 14-32 32-32h704z'/></svg>");
+		}
+		.fa-file {
+			-webkit-mask: url("data:image/svg+xml;utf8,<svg viewBox='0 0 1536 1992' xmlns='http://www.w3.org/2000/svg'><path d='M1468 380c37 37 68 111 68 164v1152c0 53-43 96-96 96H96c-53 0-96-43-96-96V96C0 43 43 0 96 0h896c53 0 127 31 164 68zm-444-244v376h376c-6-17-15-34-22-41l-313-313c-7-7-24-16-41-22zm384 1528V640H992c-53 0-96-43-96-96V128H128v1536h1280z'/></svg>");
+		}
```
Comment (laurent22) [inline ReactNativeClient/lib/renderers/noteStyle.js]: Could you please add a comment to clarify where these fonts come from and how to update them (or add new ones)?

**Comment by laurent22** [comment]
Thanks @CalebJohn, I've just left one comment and once this is done we can merge.

**Comment by tessus** [comment]
If there's zip, there should be the other compressed formats too:

.tar.gz
.tar.xz
.7z

Just my 2 cents. Or are these included in `file-archive`?

File: ReactNativeClient/lib/renderers/noteStyle.js (hunk 1 above)
Comment (CalebJohn) [reply ReactNativeClient/lib/renderers/noteStyle.js]: Done!

**Comment by CalebJohn** [comment]
@tessus tar.gz does work, but it looks like the other 2 don't. I was thinking I would probably make a pull adding some filetypes because right now what they have is very limited.

**Comment by tessus** [comment]
@CalebJohn it's not necessary for this PR. I just thought that there are way more compressed formats than just zip and it would have been nice to recognize those as well.

**Comment by laurent22** [comment]
It's all good now, many thanks for the PR @CalebJohn! For the file types we can indeed support more afterwards. I would assume they already map things like video and image mime types to the correct icons, but perhaps some are missing.

**Pull request merged**